### PR TITLE
Mapping

### PR DESCRIPTION
**Pull request by Peyman-N**

### CODE REVIEW

File: bids2openminds/mapping.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+experimental_approach={
```
Comment (lzehl) [inline bids2openminds/mapping.py:1]: either use plural or singular, but do not mix this for the different dictionaries

File: bids2openminds/mapping.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  "PC": "Phase-contrast microscopy",
+  "PLI": "Polarized-light microscopy",
+  "SEM": "Scanning electron microscopy",
+  "SPIM": "Selective plane illumination microscopy",
+  "SR": "Super-resolution microscopy",
+  "TEM": "Transmission electron microscopy",
+  "uCT": "Micro-CT",
+  "nirs": "Near-Infrared Spectroscopy ",
+  "motion": "Motion"
+}
```
Comment (lzehl) [inline bids2openminds/mapping.py:95]: remember to add a new line at the end.

File: bids2openminds/mapping.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  "meg": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/neuroimaging",
+  "eeg": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/electrophysiology",
+  "ieeg": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/electrophysiology",
+  "beh": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/behavior",
+  "pet": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/radiology",
+  "micr": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/microscopy",
+  "nirs": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/neuroimaging"
+}
+
+techniques={
```
Comment (lzehl) [inline bids2openminds/mapping.py:23]: I assume that those who are not mapped we do miss at the moment in openMINDS? I can put those on the agenda for the next ontology meeting.

**Review by lzehl — CHANGES_REQUESTED** [review]

File: bids2openminds/mapping.py (hunk 1 above)
Comment (lzehl) [reply bids2openminds/mapping.py]: I think we should also follow the constant convention in python here and identify that this is a map:

- MAP_2_EXPERIMENTAL_APPROACHES or
- MAP_TO_EXPERIMENTAL_APPROACHES

File: bids2openminds/mapping.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
+experimental_approaches={
```
Comment (lzehl) [inline bids2openminds/mapping.py:1]: Another suggestion: use the convention to constances in python and add map to distinguish between openminds property names in python module.

Suggestion: MAP_2_EXPERIMENTAL_APPROACHES or MAP_TO_EXPERIMENTAL_APPROACHES

File: bids2openminds/mapping.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
+experimental_approaches={
+  "func": "@id: https://openminds.ebrains.eu/instances/experimentalApproach/neuroimaging",
```
Comment (lzehl) [inline bids2openminds/mapping.py:2]: we discussed that this should always be a list

File: bids2openminds/mapping.py [hunk 6]
```diff
@@ -1,103 +1,120 @@
...
+  "m":["@id: https://openminds.ebrains.eu/instances/biologicalSex/male"], 
+  "M":["@id: https://openminds.ebrains.eu/instances/biologicalSex/male"], 
+  "MALE":["@id: https://openminds.ebrains.eu/instances/biologicalSex/male"], 
+  "Male":["@id: https://openminds.ebrains.eu/instances/biologicalSex/male"],
+  "female":["@id: https://openminds.ebrains.eu/instances/biologicalSex/female"],
+  "f":["@id: https://openminds.ebrains.eu/instances/biologicalSex/female"], 
+  "F":["@id: https://openminds.ebrains.eu/instances/biologicalSex/female"], 
+  "FEMALE":["@id: https://openminds.ebrains.eu/instances/biologicalSex/female"], 
+  "Female":["@id: https://openminds.ebrains.eu/instances/biologicalSex/female"],
+  "other":["@id: https://openminds.ebrains.eu/instances/biologicalSex/notDetectable"],
```
Comment (lzehl) [inline bids2openminds/mapping.py:113]: other != not detectable !!!

**Comment by lzehl** [comment]
@Peyman-N this one can be closed no? we would do a new one on the merged code now, correct?

**Comment by Peyman-N** [comment]
> @Peyman-N this one can be closed no? we would do a new one on the merged code now, correct?

I think we can, but before doing that, I am not sure I have satisfied all of your comments here. So let me check and I would close after I am sure that all of your comments have been addressed.

**Pull request closed without merging**